docs/design_docs/permission_acls.md

```diff
@@ -0,0 +1,30 @@
+# Design for Permissions and ACLs for Mazenet.
+
+WORK IN PROGRESS
+
+Author: Samuel Davidson
+
+[Typescript API](/common/api/permissions.ts)
+
+Key things it should have
+
+- Hierarchical, if ACLs for an action are not set locally, walk up the inheritance tree (mostly parent rooms) until an ACL is found. The root room should have some default ACL for everything. 
+- Room Permissions:
+  - Hierarchical, each room has an ACL for permissions 
+  - Room-data permissions E.g. who can "edit styles", "edit title"
+  - Common element permissions. E.g. Who can create a tunnel, who can 
+  - If undefined for an action, walks up the tree of parent rooms.
+- Element-unique permissions:
+  - These are set per-element. Each element has its own ACL.
+  - The permissions within here would be unique to the type of element.
+    - E.g. a tunnel element could have a "enter tunnel" element.
+    - E.g. a hypothetical "message board" element could have a "write message" permission. 
+  - These permissions don't have inhertiance. They are set per-element and don't inherit from anywhere. If they are unset then it should fail open.
```

Comment:
I think it would make sense to treat the element-specific actions the same as any other permissions i.e. they still follow normal inheritance rules. Maybe we just use a simple naming scheme to establish scope: `room.editStyles`, `structure.editPosition`, `structure.tunnel.enter`. Then you can easily set the default behavior for all of the elements in one room, change them all at once with only one change to the ACL, etc. I think this would keep the implementation a little simpler as well.